### DISCUSSION TITLE
Getting a 404 for similar routes in fastify@3

### DISCUSSION THREAD

**Discussion by oskarkivra** [fastify/fastify#4291] [Q&A] [answered]
Hi,

Let's say I have the following code:
```js
import Fastify from "fastify";

const fastify = Fastify();

fastify.get("/v1/user/:userKey", (request, reply) => {
  reply.send({ route: "2" });
});

fastify.get("/v1/:actorType/:userKey/content", (request, reply) => {
  reply.send({ route: "1" });
});

fastify.listen({ port: 3000 }, (err, address) => {
  if (err) {
    fastify.log.error(err);
    process.exit(1);
  }
  console.log(`Server is now listening on ${address}`);
});
```

If I install `fastify@3.29.2` (latest 3.x version) and try to access the second route: `curl 'http://localhost:3000/v1/user/92200'` I get back a 404. If I upgrade fastify to 4.x it works fine. So I guess this has something to do with the bump of `find-my-way` to `v6`. Is this behavior configurable in `fastify@3` or is the only option to support the code above to upgrade to `v4`?

I'm currently unable to upgrade to `v4` since we are using [Apollo server](https://github.com/apollographql/apollo-server/issues/6757#issuecomment-1201835831).

**Comment by mcollina** [comment, marked as answer]
Unfortunately no, this was fixed in the find-my-way v6.

**Reply by oskarkivra** [reply]
Okay, thanks for the quick answer.

**Reply by michael-azogu** [reply]
> Okay, thanks for the quick answer.

you can patch the package in that version

**Reply by oskarkivra** [reply]
@michael-azogu you mean [override](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides) `find-my-way` to v6? Is that safe?

**Reply by michael-azogu** [reply]
there's a tool called patch-package check it out.

**Reply by oskarkivra** [reply]
Sure but is it safe to just replace `find-my-way@4` with `find-my-way@6` and expect fastify just to work as usual? It sounds risky(?)

**Reply by michael-azogu** [reply]
don't know. you can try it in an isolated copy repo

**Reply by Eomm** [reply]
I think it is safe if you deal with the breaking changes.

Fastify v3 can use that version, but we can't release it because it _could_ break an application already in production. So if it is a fresh app, go for it
